<a href="https://colab.research.google.com/github/sehan25/ESAA/blob/main/0502_Python_Machine_Learning_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Chapter06. 피마 인디언 당뇨병 예측**

- 피마 인디언 당요녕 데이터 세트: 북아메리카 피마 지역 원주민의 Type-2 당뇨병 결과 데이터
    - Pregnancies: 임신 횟수
    - Glucose: 포도당 부하 검사 수치
    - BloodPressure: 혈압(mm Hg)
    - SkinThickness: 팔 삼두근 뒤쪽의 피하지방 측정값(mm)
    - Insulin: 혈청 인슐린(mm U/ml)
    - BMI: 체질량지수(체중(kg)/키(m))^2)
    - DiabetesPedigreeFunction: 당뇨 내력 가중치 값
    - Age: 나이
    - Outcome: 클래스 결정 값(0또는1)

* 필요한 모듈을 임포트하고, Outcome 클래스 결정값의 분포와 데이터를 개략적으로 확인 

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, precision_score, recall_score, roc_auc_score
from sklearn.metrics import f1_score, confusion_matrix, precision_recall_curve, roc_curve
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

In [3]:
from google.colab import drive
drive.mount('/content/drive')
diabetes_data = pd.read_csv('/content/drive/MyDrive/ESAA/data/diabetes.csv')
print(diabetes_data['Outcome'].value_counts())
diabetes_data.head(3)

Mounted at /content/drive
0    500
1    268
Name: Outcome, dtype: int64


Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1

    => 768개의 데이터 중에서 Negative 값 0이 500개, Positive 값 1이 268개로 Negative가 상대적으로 많음.